pkg/api/handler/helpers.go

```diff
@@ -123,3 +123,28 @@ func convertMerkleRootsRequest(beefMerkleRoots []beef.MerkleRootVerificationRequ
 
 	return merkleRoots
 }
+
+func findLastStatus(lastTxId string, statuses []*metamorph.TransactionStatus) *metamorph.TransactionStatus {
+	for _, status := range statuses {
+		if status.TxID == lastTxId {
+			return status
+		}
+	}
+	return nil
+}
+
+func filterStatusesOfInterest(txIds []string, allStatuses []*metamorph.TransactionStatus) []*metamorph.TransactionStatus {
```

Comment:
I'd rename this function to `filterStatusesByTxIDs(txIds []string, txStatuses []*metamorph.TransactionStatus)`